### PR TITLE
Venue routes and data added

### PR DESCRIPTION
**Pull request by tejasraibagi26**
### **Added 4 routes:**

**1. getAllVenues 
2. getVenueById
3. createNewVenue
4. searchVenue**

Also added errorHandling in a separate file: Errors/errorHandler.js ~ Update this file accordingly to make common error checks for all routes and data modules.

### CODE REVIEW

File: routes/venues.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      venueTimings
+    );
+
+    res.status(200).json(createVenue);
+  } catch (error) {
+    res.status(500).json({ err: error });
+  }
+});
+
+router.post("/search", async (req, res) => {
```
Comment (troytomasch) [inline routes/venues.js:102]: Is this supposed to be a get request instead?

File: Errors/errorHandler.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const { ObjectId } = require("mongodb");
+
+const checkIfExists = (array) => {
+  array.forEach((element) => {
+    if (element === undefined || element === null)
+      throw `${element} not defined`;
+  });
+};
+
+const checkIfIsString = (array) => {
```
Comment (troytomasch) [inline Errors/errorHandler.js:10]: Could we change the name to specify that it is checking the element of an array? Something like checkIfElementsAreStrings or something?

File: Errors/errorHandler.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  array.forEach((element) => {
+    if (typeof element !== "string") throw `${element} not of type string`;
+  });
+};
+
+const checkIfValidObjectId = (id) => {
+  if (!ObjectId.isValid(id)) throw "Incorrect Object Id";
+};
+
+const checkIfNotEmptyString = (array) => {
```
Comment (troytomasch) [inline Errors/errorHandler.js:20]: Same thing as the other comment with a clearer name

File: data/venue.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  return allVenues;
+};
+
+const getVenueById = async (id) => {
+  //Error Handling
+  const array = [id];
+  errorHandler.checkIfExists(array);
+  errorHandler.checkIfIsString(array);
+  errorHandler.checkIfValidObjectId(id);
+  errorHandler.checkIfNotEmptyString(array);
```
Comment (troytomasch) [inline data/venue.js:22]: Switch these to make errors more accurate

**Review by troytomasch — CHANGES_REQUESTED** [review]

File: routes/venues.js (hunk 1 above)
Comment (tejasraibagi26) [reply routes/venues.js:102]: We could make it a get request. I'm currently passing the search key via a post request to the backend. If we make it get we need to send it via the url, something like this: /search?q="search_string". But i feel making it post will be better.

File: Errors/errorHandler.js (hunk 2 above)
Comment (tejasraibagi26) [reply Errors/errorHandler.js:10]: Sure. I'll make the changes and push the changes.

File: routes/venues.js (hunk 1 above)
Comment (troytomasch) [reply routes/venues.js:102]: Just seemed off to me for a post request to be searching for data but I'm good with leaving it the way it is

**Review by troytomasch — APPROVED** [review]
I'm good to merge it. Do we want another review or should we just merge it?

**Pull request merged**